partitions.go

```diff
@@ -293,3 +293,31 @@ func AssignFreshPartitionSpecIDs(spec *PartitionSpec, old, fresh *Schema) (Parti
 
 	return NewPartitionSpec(newFields...), nil
 }
+
+func GeneratePartitionFieldName(schema *Schema, field PartitionField) (string, error) {
+	sourceName, exists := schema.FindColumnName(field.SourceID)
+	if !exists {
+		return "", fmt.Errorf("could not find field with id %d", field.SourceID)
+	}
+	transform := field.Transform
+	switch t := transform.(type) {
+	case IdentityTransform:
+		return sourceName, nil
+	case BucketTransform:
+		return fmt.Sprintf("%s_bucket_%d", sourceName, t.NumBuckets), nil
+	case TruncateTransform:
+		return fmt.Sprintf("%s_trunc_%d", sourceName, t.Width), nil
+	case YearTransform:
+		return sourceName + "_year", nil
+	case MonthTransform:
+		return sourceName + "_month", nil
+	case DayTransform:
+		return sourceName + "_day", nil
+	case HourTransform:
+		return sourceName + "_hour", nil
```

Comment:
all of these could likely be replaced by calling `.String()` on the transform:

```go
switch t := transform.(type) {
case IdentityTransform:
        return sourceName, nil
case VoidTransform:
        return sourceName + "_null", nil
default:
        return sourcecName + "_" + t.String(), nil
}
```

Reply:
I'm okay with using those as partition field names, is there an existing example of the generated names in the java implementation?

Reply:
damn, okay i guess we have to leave those separately so we can be consistent with the others. 

Can you add a comment linking to the java impl so we remember why we're not just using the `.String()` method?